### ISSUE TITLE
Nearpod 

### ISSUE THREAD

**Issue by Jadoo4QFan** [yt-dlp/yt-dlp#5090]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

USA

### Example URLs

https://nearpod.com/collaborate/reports/e9e8e2a895dec0ba6c9c850a3c866a2f/421747216?&parent_id
=0&hideStudentNames=1

### Provide a description that is worded well enough to be understood

Please add support for Nearpod. It has these collaborative boards where users can post videos and audios. Please add support for downloading video and audio from the site.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[Documents]$ yt-dlp -vU https://nearpod.com/col
laborate/reports/e9e8e2a895dec0ba6c9c850a3c866a
2f/421747216?&parent_id=0&hideStudentNames=1
[debug] Command-line config: ['-vU', 'https://n
earpod.com/collaborate/reports/e9e8e2a895dec0ba
6c9c850a3c866a2f/421747216?&parent_id=0&hideStu
dentNames=1']
[debug] Encodings: locale UTF-8, fs utf-8, pref
 UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (pi
p)
[debug] Python 3.9.10+ (CPython 64bit) - macOS-
16.0-iPhone14,5-arm-64bit 
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg n5.0.1-4-ga5ebb3d2
5e (setts), ffprobe n5.0.1-4-ga5ebb3d25e, phant
omjs present, rtmpdump present
[debug] Optional libraries: Cryptodome-3.15.0, 
certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [generic] Extracting URL: https://nearpod.com/collaborate/reports/e9e8e2a895dec0ba6c9c850a3c866a2f/421747216?&parent_id=0&hideStudentNames=1
[generic] 421747216?&parent_id=0&hideStudentNames=1: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 421747216?&parent_id=0&hideStudentNames=1: Extracting information
[debug] Looking for Brightcove embeds
[debug] Looking for embeds
ERROR: Unsupported URL: https://nearpod.com/collaborate/reports/e9e8e2a895dec0ba6c9c850a3c866a2f/421747216?&parent_id=0&hideStudentNames=1
Traceback (most recent call last):
  File "/var/mobile/Containers/Data/Application/BE92A072-84F0-4DB1-8FB4-0C9E11A11159/Library/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1459, in wrapper
    return func(self, *args, **kwargs)
  File "/var/mobile/Containers/Data/Application/BE92A072-84F0-4DB1-8FB4-0C9E11A11159/Library/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1535, in __extract_info
    ie_result = ie.extract(url)
  File "/var/mobile/Containers/Data/Application/BE92A072-84F0-4DB1-8FB4-0C9E11A11159/Library/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 670, in extract
    ie_result = self._real_extract(url)
  File "/var/mobile/Containers/Data/Application/BE92A072-84F0-4DB1-8FB4-0C9E11A11159/Library/lib/python3.9/site-packages/yt_dlp/extractor/generic.py", line 3078, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://nearpod.com/collaborate/reports/e9e8e2a895dec0ba6c9c850a3c866a2f/421747216?&parent_id
=0&hideStudentNames=1
```

**Comment by coletdjnz** [comment]
Request to `https://collaborative-api.nearpod.com/sessions/421747216/e9e8e2a895dec0ba6c9c850a3c866a2f?isReport=1` to get board details and cards.

The cards in above request do not contain information about their assets. Just the assetUid (if there is any linked assets). To get the asset information it goes something like:
1. For every assetUid, request `/assets/<assetUid>/type` for every card to see if it is a video, audio, image etc.
2. Request the asset metadata:
    - If the asset is audio, you need to request `/assets/<assetUid>/audios/status`
   -  If asset is video, you need to request `/assets/<assetUid>/status`
   - The above contain the metadata you want from yt-dlp - formats, title, duration, etc.
   - Be aware of the different types of video sources - e.g. youtube, amazon(direct link?)
   - The API suggests it keeps track of different versions

In other words, a lot of requests ~~(what a great API)~~

Maybe someone can figure something out to return relevant data in less requests.
